zilliqa/src/pool.rs

```diff
@@ -132,6 +132,43 @@ impl TransactionPool {
         }
     }
 
+    /// Returns a list of txns that are pending for inclusion in the next block
+    pub fn pending_hashes(&self) -> Vec<Hash> {
+        let mut pending_hash = Vec::<Hash>::new();
+        let mut pending_set = HashSet::new();
+        let mut ready = self.ready.clone();
```

Comment:
Haven't we kept the transaction pool cloning here but hidden it away? @bzawisto might have a good point that if we use a different data structure here we can remove the clone (as ZQ1 seems to manage?)?

Reply:
In this case, it's just cloning the ready heap, which is relatively small as compared to the actual list of transactions. But if the consensus is to rewrite the entire pool using a different data structure, then I'll take a look into that.

Reply:
If @bzawisto doesn't disagree, I think it would be fine to merge this for now and look at changing the data structure afterwards?

Reply:
That's good too. @bzawisto if you have no objections, could you approve this PR?